course_info/static/course_info/js/controllers/PeopleController.js

```diff
@@ -14,110 +14,157 @@
         };
 
         // set up functions we'll be calling later
-        $scope.addUser = function(searchTerm) {
-            $scope.searchInProgress = true;
-            if ($scope.searchResults.length === 0) {
-                $scope.lookup(searchTerm);
-            }
-            else if ($scope.searchResults.length === 1) {
-                $log.error('Add user button pressed while we have a single search result');
+        $scope.addNewMember = function(personResult, members) {
+            /* Make a call to add the person to the course as a new member
+             if person is:
+             - not already in `members`
+             - not found in people lookup
+             - represented by more than one profile
+             Returns a promise representing the call made to add the new member
+             to the course, or null if the add was not attempted for one of the
+             reasons listed above.
+             */
+            var personRecords = personResult[0];
+            var searchTerm = personResult[1];
+            var filteredResults = $scope.filterSearchResults(personRecords);
+            if (filteredResults.length == 1) {
+                var memberRecordsInCourse = members[filteredResults[0].univ_id];
+                if (angular.isUndefined(memberRecordsInCourse)) {
+                    var name = $scope.getProfileFullName(filteredResults[0]);
+                    var postParams = {
+                        user_id: filteredResults[0].univ_id,
+                        role_id: $scope.selectedRole.roleId};
+                    return $scope.addNewMemberToCourse(postParams, name,
+                            searchTerm);
+                } else {
+                    // the user already has an enrollment in the course
+                    $scope.messages.warnings.push({
+                        type: 'alreadyInCourse',
+                        // just pick the first enrollment to find the name
```

Comment:
This is picking from the first profile match, not the first enrollment in this course.  Probably we want the latter.